.github/workflows/main.yml

```diff
@@ -0,0 +1,129 @@
+name: Build and Upload story Binary
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  # TESTING ONLY
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  id-token: write
+  contents: write
+  pull-requests: write
+  actions: write
+
+env:
+  NUM_INTERNAL_BINARIES_TO_KEEP: 50
+  NUM_PUBLIC_BINARIES_TO_KEEP: 400
+  S3_BUCKET: story-geth-binaries
+  BIN_NAME: story
+
+jobs:
+  Timestamp:
+    uses: storyprotocol/gha-workflows/.github/workflows/reusable-timestamp.yml@main
+  
+  build_and_push:
+    needs: Timestamp
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [linux-amd64]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+
+      - name: Authenticate to GCP
+        uses: google-github-actions/auth@v1
+        with:
+          workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER_REPO_STORY }}
+          service_account: ${{ secrets.GCP_SERVICE_ACCOUNT_REPO_STORY }}
+
+      - name: Set up Google Cloud SDK
+        uses: google-github-actions/setup-gcloud@v1
```

Comment:
```suggestion
        uses: google-github-actions/setup-gcloud@v2
```

Reply:
Hey @AndyBoWu , would it be cool to bump `google-github-actions/setup-gcloud@v1 to v2`? `v1` is using the deprecated Node 16, which could a possible security concern.